research-prototype/PATCG_2022.8.11_README.md

```diff
@@ -0,0 +1,143 @@
+
+# PATCG 8/11 Benchmarks
+
+Here are the run commands for the benchmarks we shared of IPA the the 8/11/22 PATCG. 
+
+We ran with the 3 party honest majority malicious protocol  `sy-rep-ring-party.x` with the default statistical security of 40 bits. For semi-honest we ran replicated secret sharing protocol ‘ring.sh’
+
+To enable mod 2^k ring computation with custom k, add `MOD = -DRING_SIZE=32` to `MP-SPDZ/CONFIG.mine` and rebuild with `make -B -j8 sy-rep-ring-party.x’ .  All of our results are were for k = 32. 
+
+The version of the code we used is called `vectorized.mpc` and differs in how it takes arguments than the `ipae2e` version on the current main branch of raw-ipa.  The particular commit we benchmarked was [https://github.com/bmcase/raw-ipa/commit/8432970926cecef90da3e2bf6202577950d887d6](https://github.com/bmcase/raw-ipa/commit/8432970926cecef90da3e2bf6202577950d887d6) though we’ve cleaned up the comments in `vectorized.mpc` a bit since.
+
+# Scaling # rows
+
+
+## 2^12 4
+
+
+
+* `/usr/bin/time --verbose ../../../MP-SPDZ/compile.py -C -R 32 vectorized 12`
+* P0:  `/home/ec2-user/workspace/MP-SPDZ/Scripts/../sy-rep-ring-party.x --player 0 vectorized-12 --hostname ec2-35-161-7-220.us-west-2.compute.amazonaws.com`
```

Comment:
we should stub out these hostnames and put them into a environmental variable so that folks can replicate.